.github/workflows/test.yml

```diff
@@ -0,0 +1,64 @@
+name: Test
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: Wake 0.18 scala compile
+    runs-on: ubuntu-latest
+    env:
+      # $WIT_WORKSPACE path is relative to $GITHUB_WORKSPACE
+      WIT_WORKSPACE: wit-workspace
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        path: wit-packages/ci-root-package
+        # Fetch all history and tags
+        fetch-depth: 0
+
+    - name: 'Wit Init'
+      uses: sifive/wit/actions/wit@v0.13.1
+      with:
+        command: init
+        arguments: |
+          ${{ env.WIT_WORKSPACE }}
+          -a ./wit-packages/ci-root-package
+        force_github_https: true
+
+    - name: Run wake-based scala build
+      working-directory: ${{ env.WIT_WORKSPACE }}
+      run: |
+        set -euxo pipefail
+        # Take back ownership of wit workspace, since it was created while
+        # under a Docker container and permissions default to root.
+        sudo chown -R $(id -u):$(id -g) .
+        # Install protobuf
+        mkdir protoc
+        cd protoc
+        wget https://github.com/protocolbuffers/protobuf/releases/download/v3.7.1/protoc-3.7.1-linux-x86_64.zip
```

Comment:
You might want to replace most of the set up with environment-blockci-sifive, since that sets up the all the dependencies needed to run most of the build rules defined in api-generator-sifive, will allow you to exercise more CI tests than just compiling the Scala module. It also obviates the `chown` workaround by continuing to run in Docker and therefore under root.

But this LGTM as is.